cvat-sdk/cvat_sdk/core/proxies/tasks.py

```diff
@@ -156,7 +157,7 @@ def upload_data(
                     break
 
                 elif status.value == models.RequestStatus.allowed_values[("value",)]["FAILED"]:
-                    raise exceptions.ApiException(status=status, reason=message, http_resp=response)
+                    raise BackgroundRequestException(message)
```

Comment:
Consider replacing this code block with a `self.client.wait_for_completion()` call and making this change in that function instead.

Reply:
I thought about this, but did not want to refactor this code in the current PR. Okay, I'll do it.